lib/claim_documents/monitor.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'zero_silent_failures/monitor'
+
+module ClaimDocuments
+  ##
+  # Monitor functions for Rails logging and StatsD
+  # @todo abstract, split logging for controller and sidekiq
+  #
+  class Monitor < ::ZeroSilentFailures::Monitor
+    # statsd key for document uploads
+    DOCUMENT_STATS_KEY = 'api.document_upload'
+
+    def track_document_upload_attempt(form_id, current_user)
+      StatsD.increment("#{DOCUMENT_STATS_KEY}.attempt", tags: ["form_id: #{form_id}"])
+      Rails.logger.info("Creating PersistentAttachment FormID=#{form_id}",
+                        { user_account_uuid: current_user&.user_account_uuid,
+                          statsd: "#{DOCUMENT_STATS_KEY}.attempt" })
+    end
+
+    def track_document_upload_success(form_id, attachment_id, current_user)
+      StatsD.increment("#{DOCUMENT_STATS_KEY}.success", tags: ["form_id: #{form_id}"])
+      Rails.logger.info("Success creating PersistentAttachment FormID=#{form_id} AttachmentID=#{attachment_id}",
+                        { attachment_id:, user_account_uuid: current_user&.user_account_uuid,
+                          statsd: "#{DOCUMENT_STATS_KEY}.success" })
+    end
+
+    def track_document_upload_failed(form_id, attachment_id, current_user, e)
+      StatsD.increment("#{DOCUMENT_STATS_KEY}.failure", tags: ["form_id: #{form_id}"])
+      log_silent_failure({ form_id: form_id, attachment_id: attachment_id }, current_user&.user_account_uuid)
```

Comment:
Is this a silent failure? It notifies the user synchronously, right?

Reply:
I think this would be considered a silent failure, since submission failures due to attachment failures aren't properly being logged. And no, the similar `log_silent_failure_avoided` function is the one that sends the email to the user when the parameter `email_confirmed: true` is passed.

Reply:
The upload error is/should be displayed to the user though.  so this should not be a silent failure.
Also, there should be no space in the 'tags' values, and you should be able to use `Logging::Monitor#track_request`

Reply:
The upload error is/should be displayed to the user though. so this should not be a silent failure.
Also, there should be no space in the 'tags' values, and you should be able to use Logging::Monitor#track_request